cuda_bindings/tests/run_python_code_safely.py

```diff
@@ -0,0 +1,86 @@
+import multiprocessing
+import queue  # for Empty
+import subprocess  # nosec B404
+import sys
+import traceback
+from io import StringIO
+
+
+class Worker:
+    def __init__(self, python_code, result_queue):
+        self.python_code = python_code
+        self.result_queue = result_queue
+
+    def __call__(self):
+        # Capture stdout/stderr
+        old_stdout = sys.stdout
+        old_stderr = sys.stderr
+        sys.stdout = StringIO()
+        sys.stderr = StringIO()
+
+        try:
+            exec(self.python_code, {"__name__": "__main__"})  # nosec B102
+            returncode = 0
+        except SystemExit as e:  # Handle sys.exit()
+            returncode = e.code if isinstance(e.code, int) else 0
+        except BaseException:
+            traceback.print_exc()
+            returncode = 1
+        finally:
+            # Collect outputs and restore streams
+            stdout = sys.stdout.getvalue()
+            stderr = sys.stderr.getvalue()
+            sys.stdout = old_stdout
+            sys.stderr = old_stderr
+            try:  # noqa: SIM105
+                self.result_queue.put((returncode, stdout, stderr))
+            except Exception:  # nosec B110
+                # If the queue is broken (e.g., parent gone), best effort logging
+                pass
+
+
+def run_python_code_safely(python_code, *, timeout=None):
```

Comment:
Can we rename this function? Something like `run_py_spawned_subprocess` or something like that?

Reply:
Done under #621